Client/src/Unified/Services/Submitter/Service.cs

```diff
@@ -284,194 +277,141 @@ public string SessionId
-                                                             CancellationToken.None)
-                                          .Result);
-          }
-          catch (Exception e)
+          var resultStatusCollection = SessionService.GetResultStatus(chunk);
+
+          foreach (var resultStatusData in resultStatusCollection.IdsReady)
           {
-            Logger?.LogWarning(e,
-                               "Response handler for {taskId} threw an error",
-                               resultStatusData.TaskId);
+            var taskId = resultStatusData.TaskId;
+            fetched += 1;
+
             try
             {
-              errorHandler(resultStatusData.TaskId,
-                           TaskStatus.Error,
-                           e.Message + e.StackTrace);
+              var byteResult = SessionService.TryGetResultAsync(new ResultRequest
+                                                                {
+                                                                  ResultId = resultStatusData.ResultId,
+                                                                  Session  = SessionId,
+                                                                },
+                                                                CancellationToken.None)
+                                             .Result;
+              Logger?.LogTrace("Response handler for {taskId}",
+                               resultStatusData.TaskId);
+
+              var result = ProtoSerializer.DeSerializeMessageObjectArray(byteResult);
+              ResultHandlerDictionary[taskId]
+                .HandleResponse(result?[0],
+                                taskId);
             }
-            catch (Exception e2)
+            catch (Exception e)
             {
-              Logger?.LogError(e2,
-                               "An error occured while handling another error: {details}",
-                               e);
+              Logger?.LogDebug(e,
+                               "Response handler for {taskId} threw an exception, calling error handler",
+                               resultStatusData.TaskId);
+
+              var ex = e;
+
+              if (e is AggregateException ae)
+              {
+                ex = ae.InnerExceptions.Count > 1
+                       ? ae
+                       : ae.InnerException;
+              }
+
+              try
+              {
+                ResultHandlerDictionary[taskId]
+                  .HandleError(new ServiceInvocationException(ex,
+                                                              ArmonikStatusCode.Unknown),
+                               taskId);
+              }
+              catch (Exception e2)
+              {
+                Logger?.LogError(e2,
+                                 "An exception was thrown while handling a previous exception in the result handler of {taskId}",
+                                 taskId);
```

Comment:
Re-throw exception here. User process should stop here since they don't manage properly the exception